mlrun/package/context_handler.py

```diff
@@ -0,0 +1,355 @@
+
+        # If a context was found, give the packagers manager custom packagers to collect (if available):
+        if self._context is not None:
+            # TODO: Complete this in the project spec first
+            # self._packagers_manager.collect_packagers(packagers=self._context.project.get_custom_packagers())
+            pass
+
+    def is_context_available(self) -> bool:
+        """
+        Check if a context was found by the method `look_for_context`.
+
+        :returns: True if a context was found and False otherwise.
+        """
+        return self._context is not None
+
+    def parse_inputs(
+        self,
+        args: tuple,
+        kwargs: dict,
+        type_hints: OrderedDict,
+    ) -> tuple:
+        """
+        Parse the given arguments and keyword arguments data items to the expected types.
+
+        :param args:       The arguments tuple passed to the function.
+        :param kwargs:     The keyword arguments dictionary passed to the function.
+        :param type_hints: An ordered dictionary of the expected types of arguments.
+
+        :returns: The parsed args (kwargs are parsed inplace).
+        """
+        # Parse the type hints (in case some were given as strings):
+        type_hints = {
+            key: self.parse_type_hint(type_hint=value)
+            for key, value in type_hints.items()
+        }
+
+        # Parse the arguments:
+        parsed_args = []
+        type_hints_keys = list(type_hints.keys())
+        for i, argument in enumerate(args):
+            if isinstance(argument, DataItem) and type_hints[
+                type_hints_keys[i]
+            ] not in [
+                inspect._empty,
+                DataItem,
+            ]:
+                parsed_args.append(
+                    self._packagers_manager.unpack(
+                        data_item=argument,
+                        type_hint=type_hints[type_hints_keys[i]],
+                    )
+                )
+                continue
+            parsed_args.append(argument)
+        parsed_args = tuple(parsed_args)  # `args` is expected to be a tuple.
+
+        # Parse the keyword arguments:
+        for key in kwargs.keys():
+            if isinstance(kwargs[key], DataItem) and type_hints[key] not in [
+                inspect._empty,
+                DataItem,
+            ]:
+                kwargs[key] = self._packagers_manager.unpack(
+                    data_item=kwargs[key], type_hint=type_hints[key]
+                )
+
+        return parsed_args
+
+    def log_outputs(
+        self,
+        outputs: list,
+        log_hints: List[Union[Dict[str, str], str, None]],
+    ):
+        """
+        Log the given outputs as artifacts with the stored context.
+
+        :param outputs:   List of outputs to log.
+        :param log_hints: List of log hints (logging configurations) to use.
+        """
+        # Go over the outputs and pack them:
+        for obj, log_hint in zip(outputs, log_hints):
```

Comment:
this one is still work in progress, I don't want to raise any issues (perhaps I should add a warning) simply skip them so this behavior is good.

What I want to add is support for unknown amount of returning values from a function (can be a list of stuff to log separately or a dictionary to log its items separately). For example, log hint like this: `"**guy_: plot"` and an out of a dictionary: `a: Dict[str, plotly.Figure] = {"a_plot": ..., "another": ..., ...}` will log all dictionary items as plot artifacts using the prefix `"guy_"` so you can expect these artifacts at the end of the run: `["guy_a_plot", "guy_another"]`.

Reply:
Not sure I understand why now you want to ignore mismatches. Note that if for example the hints are shorter than return values, then the last return values will just not get processed by this since it will not be part of the `zip` result.

Reply:
I won't ignore them, I'll add a warning that some of the values won't be logged as log hints are missing. I can add something like default key and enumeration to log whatever, which seems like a good idea, but how would you turn it off? Maybe passing `returns=False` will tell to ignore all and `returns=True` will use the default keys and if given specific then it will ignore if log hints are not at the correct length with a warning.

By default keys, imagine the func is:

```python
def my_handler():
    ...
    return 10, 20, my_df, another_df, "hello"
```

So the logged outputs will be: `["result", "result_2", "dataset", "dataset_2", "result_3"]`